src/tiny.ts

```diff
@@ -75,38 +77,49 @@ export class TinyPg {
       )
    }
 
-   async query<T extends object = any, P extends object = T.TinyPgParams>(raw_sql: string, params?: P): Promise<T.Result<T>> {
+   query<T extends object = any, P extends object = T.TinyPgParams>(raw_sql: string, params?: P): Promise<T.Result<T>> {
+      const query_id = Uuid.v4()
+      const hook_lifecycle = this.makeHooksLifeCycle()
+      const [new_query, new_params] = hook_lifecycle.preRawQuery({ query_id: query_id }, [raw_sql, params]).args
+
       return Util.stackTraceAccessor(this.options.capture_stack_trace, async () => {
          const parsed = P.parseSql(raw_sql)
 
          const db_call = new DbCall({
             name: 'raw_query',
             key: null,
-            text: raw_sql,
+            text: new_query,
             parameterized_query: parsed.parameterized_sql,
             parameter_map: parsed.mapping,
             prepared: false,
          })
 
-         return await this.performDbCall<T>(db_call, params)
+         return await this.performDbCall<T>(db_call, hook_lifecycle, new_params, query_id)
       })
    }
 
-   async sql<T extends object = any, P extends object = T.TinyPgParams>(name: string, params?: P): Promise<T.Result<T>> {
+   sql<T extends object = any, P extends object = T.TinyPgParams>(name: string, params?: P): Promise<T.Result<T>> {
+      const query_id = Uuid.v4()
+      const hook_lifecycle = this.makeHooksLifeCycle()
+      const [new_name, new_params] = hook_lifecycle.preSql({ query_id: query_id }, [name, params]).args
+
+      // TODO: Handle caller_context for transactions calling this function
+      // const preSqlResult = this.runPreSqlHooks(name, query_id, params, null)
+
       return Util.stackTraceAccessor(this.options.capture_stack_trace, async () => {
-         log('sql', name)
+         log('sql', new_name)
 
-         const db_call: DbCall = this.sql_db_calls[name]
+         const db_call: DbCall = this.sql_db_calls[new_name]
```

Comment:
this probably needs to be original name for the lookup. Needs a test for name transformation